### PR TITLE
Added Reloading Functionality, Hot-Reloading.

### PR DESCRIPTION
**Pull request by TejasJD**
- Script files can now be added and built independently even when working on an Active Horizon Game Project.
- Upon compiling the script project, the Game Project automatically detects changes and reloads the scripts.
- Switching between different projects, load scripts from the newly switched project.

### CODE REVIEW

**Comment by Minghui-Gao** [comment]
This looks very good, no need to compile the script file separately to generate the dll

**Comment by TejasJD** [comment]
> This looks very good, no need to compile the script file separately to generate the dll

Yeah, you don't have to close the engine in any case atleast.

**Comment by TejasJD** [comment]
Hot Reloading is possible now. When the Script Project is compiled, the changes are automatically detected and picked up by the Editor.

**Review by alex-mulkerrins — APPROVED** [review]

File: HorizonEngine/src/HorizonEngine/Scripting/ScriptEngine.cpp [hunk 1]
```diff
@@ -14,51 +17,25 @@ namespace Hzn
...
-			*outSize = size;
-			return buffer;
-		}
-
 		static MonoAssembly* LoadCSharpAssembly(const std::string& assemblyPath)
 		{
 			uint32_t fileSize = 0;
-			char* fileData = Utils::ReadBytes(assemblyPath, &fileSize);
 
 			// NOTE: We can't use this image for anything other than loading the assembly because this image doesn't have a reference to the assembly
```
Comment (LiamANeeson) [inline HorizonEngine/src/HorizonEngine/Scripting/ScriptEngine.cpp:36]: Why is an image required?

**Review by LiamANeeson — APPROVED** [review]
LGTM!

**Review by Minghui-Gao — APPROVED** [review]

File: HorizonEngine/src/HorizonEngine/Scripting/ScriptEngine.cpp (hunk 1 above)
Comment (TejasJD) [reply HorizonEngine/src/HorizonEngine/Scripting/ScriptEngine.cpp:36]: Image's are what hold the intermediate code, in the right formats which is then used to access information in the DLL, or the assembly.

**Pull request merged**